src/book/03-guides/01-main-codebase/02-script-tuner.mdx

```diff
@@ -0,0 +1,99 @@
+---
+section: Guides
+chapter: Main Codebase
+title: Running and Tuning Scripts
+description: How to tune and run scripts.
+slug: /guides/main/script-tuner
+---
+
+Scripts are static motions for the robot. They specify what joint angles to move to and how long the robot should take to go to those joint positions. For example, standing up is a very basic script telling the robot to move their joints to the stand position over one second. There can be many of these position specifications in a row to make the robot do more complex movements like getting up or kicking.
+
+The Script Engine specifies how the scripts work. It takes each joint position and time from the `.yaml` file and sends the information to the servos.
+
+Information on Script Engine is on the [Motion](/system/subsystems/motion#script-engine) page and Script Runner and Script Tuner on the [Behaviour](/system/subsystems/behaviour#script-runner) page.
+
+## Script Runner
+
+The script runner command will run a script. The command for this is
+
+```bash
+./scriptrunner Script.yaml
+```
+
+Where you would replace `Script.yaml` with the name of the script. All scripts can be found in the NUbots repository under `/module/motion/ScriptEngine/data/scripts/`.
+
+The script you are running needs to be on the robot in the `/scripts/` folder. These scripts can be robot specific. For example, if you were running `Stand.yaml` on `nugus1`, the ScriptEngine would first look for `Stand.yaml` in the `/scripts/` folder, and then it would check in the `/scripts/nugus1/` folder for a `Stand.yaml` file. If it exists, it will override with that script.
+
+## Script Tuner
+
+Script tuner is an easy way to make scripts. To use script tuner, run
+
+```bash
+./scripttuner /path/to/Script.yaml
+```
+
+Where `Script.yaml` is either the name of an existing script, if you want to edit it, or the name for a new script. `/path/to/` is the path to the script. This may be `scripts/nugus/Script.yaml`, or `scripts/nugus*/Script.yaml` where `*` is the robot number for a robot specific script.
+
+You will then be able to tune the script. You will see the following interface for script editing.
+
+![First line states 'Scripts: scripts/nugus2/walk.yaml'. Next line says 'Frames:' with consecutive numbers from 1 to 44 laid out in a row. The number '1' is selected. The next line says 'Duration: 3000'. The next section is a table with the first column specifying servo names such as 'right shoulder pitch'. The second column specifies the angle of the servo. This is of the format 'Angle: +1.45'. Some of the rows have dashes instead of numbers for the angle, denoting an empty field. The third row states the gain of the servo. This is of the form 'Gain: 30.0'. Like the angle, some of these are blanked out with dashes. Empty angle and gain fields are paired. In the next section there are several keyboard commands which allow the user to use the interface.](./images/scripttuner.png)
```

Comment:
This ALT text is excellent, well done! 👏 👏 👏

Reply:
I'm a little concerned it's too lengthy. Alt text should be concise. The picture is of text, so it might be worth... making it text in NUbook. Also, I explain it afterwards anyway.

Reply:
Could probably be made shorter. And yeah, an SVG with readable text would probably work better here 👍.

As is though I think it's better to err on the side of more descriptive ALT text than shorter ALT text. Also I think the following lines describe what the parts of the screenshot are for and what they do, assuming the user is aware of (has seen or knows about) those parts; but that can probably be established with an opening sentence or two.

Reply:
I was thinking more just a code block with the text, but I'll have a look into how SVGs can be used in this case if it can be more presentable and accessible.

Reply:
A code block would probably be simpler and perhaps just as effective.